tests/test_server.py

```diff
@@ -1401,57 +1402,14 @@ async def test_send_deploy_done(server, client, environment, agent, caplog, meth
             result=executor.DeployResult(
                 rvid=rvid_r1_v1,
                 action_id=action_id,
-                status=const.ResourceState.deployed,
+                resource_state=HandlerResourceState.deployed,
                 messages=[],
                 changes={"attr1": AttributeStateChange(current="test", desired="test2")},
                 change=const.Change.created,
             )
         )
 
 
-async def test_send_deploy_done_invalid_state(server, client, environment, agent, caplog):
-    """
-    Ensure proper error handling when a transient state is passed to the `send_deploy_done` method.
-    """
-    env_id = uuid.UUID(environment)
-
-    model_version = 1
-    cm = data.ConfigurationModel(
-        environment=env_id,
-        version=model_version,
-        date=datetime.now().astimezone(),
-        total=1,
-        version_info={},
-        is_suitable_for_partial_compiles=False,
-    )
-    await cm.insert()
-
-    rvid_r1_v1 = ResourceVersionIdStr(f"std::testing::NullResource[agent1,name=file1],v={model_version}")
-    await data.Resource.new(
-        environment=env_id,
-        status=const.ResourceState.available,
-        resource_version_id=rvid_r1_v1,
-        attributes={"purge_on_delete": False, "requires": []},
-    ).insert()
-
-    action_id = uuid.uuid4()
-    update_manager = persistence.ToDbUpdateManager(client, env_id)
-    await update_manager.send_in_progress(action_id, rvid_r1_v1)
-
-    with pytest.raises(ValueError) as exec_info:
-        await update_manager.send_deploy_done(
-            result=executor.DeployResult(
-                rvid=rvid_r1_v1,
-                action_id=action_id,
-                status=const.ResourceState.deploying,
-                messages=[],
-                changes={"attr1": AttributeStateChange(current=None, desired="test")},
-                change=const.Change.created,
-            )
-        )
-    assert "No transient state can be used to mark a deployment as done" in str(exec_info.value)
-
-
```

Comment:
I removed this test because we no longer set the status of `DeployResult` like this. We use the `HandlerResourceState` which (I believe) does not have transient states

Reply:
We didn't touch the update manager, did we?

Reply:
I believe we didn't. We just changed the definition of `DeployResult`, so this test does not make sense anymore

Reply:
Ah, you mean that it has a new field now? But as far as I can tell this method doesn't use the new field directly? Perhaps we should update it.

Either way, I think you're right that the test can go, but if we don't update the method it's built on an assumption that despite it being typed as the one enum, it will really only return what's in the other.

That said, if it proves difficult (I'm assuming a one-line change now), it's probably not worth it.

Reply:
It has a new field (`resource_state`) which uses a different class (`HandlerResourceState`). We no longer support creating with the `status` field as it is read-only.
`HandlerResourceState`, by definition does not have transient states. So we can't actually call `send_deploy_done` with a transient state we would fail on creating `DeployResult` instead.
I'm not sure I understand this part:
> but if we don't update the method it's built on an assumption that despite it being typed as the one enum, it will really only return what's in the other.

Reply:
I mean, the update manager still reads `status` instead of `resource_state`, right?

Reply:
So I'm arguing that it shouldn't. If it expects only the handler states, it should read only those. That wasn't possible before but this PR gives us the oppurtunity to clean that up.